sdk/search/search-documents/test/public/utils/setup.ts

```diff
@@ -55,39 +56,37 @@ export async function createIndex(
       parameters: { metric: "euclidean" },
     },
   ];
-  await Promise.all(algorithmConfigurations.map(renameUniquelyInPlace));
+  await Promise.all(algorithmConfigurations.map(renameUniquelyInPlace("name")));
   const [hnswAlgorithmConfigurationName, exhaustiveKnnAlgorithmConfigurationName] =
     algorithmConfigurations.map((c) => c.name);
 
-  const compressionConfigurations: VectorSearchCompressionConfiguration[] = [
+  const compressionConfigurations: VectorSearchCompression[] = [
     {
-      name: "vector-search-compression-configuration",
+      compressionName: "vector-search-compression-configuration",
       kind: "scalarQuantization",
       parameters: { quantizedDataType: "int8" },
       rerankWithOriginalVectors: true,
     },
   ];
-  await Promise.all(compressionConfigurations.map(renameUniquelyInPlace));
+  await Promise.all(compressionConfigurations.map(renameUniquelyInPlace("compressionName")));
   const [scalarQuantizationCompressionConfigurationName] = compressionConfigurations.map(
-    (c) => c.name,
+    (c) => c.compressionName,
   );
 
-  const vectorSearchProfiles = [
+  const vectorSearchProfiles: VectorSearchProfile[] = [
     {
       name: "vector-search-profile",
-      vectorizer: isPreview ? azureOpenAiVectorizerName : undefined,
+      vectorizerName: isPreview ? azureOpenAiVectorizerName : undefined,
       algorithmConfigurationName: exhaustiveKnnAlgorithmConfigurationName,
     },
     {
       name: "vector-search-profile",
-      vectorizer: isPreview ? azureOpenAiVectorizerName : undefined,
+      vectorizerName: isPreview ? azureOpenAiVectorizerName : undefined,
       algorithmConfigurationName: hnswAlgorithmConfigurationName,
-      compressionConfigurationName: isPreview
-        ? scalarQuantizationCompressionConfigurationName
-        : undefined,
+      compressionName: isPreview ? scalarQuantizationCompressionConfigurationName : undefined,
     },
   ];
-  await Promise.all(vectorSearchProfiles.map(renameUniquelyInPlace));
+  await Promise.all(vectorSearchProfiles.map(renameUniquelyInPlace("name")));
```

Comment:
nit: I actually prefer something like 
```ts
Promise.all(vectorSearchProfiles.map(p => renameUniquelyInPlace("name", p))
```

I definitely understand the appeal and have had my share of "partial application/currying" equivalent in JS code but not everyone has functional background and they'd appreciate the readability!

Reply:
Fixed!